src/wui/gamedetails.cc

```diff
@@ -172,82 +79,126 @@ void GameDetails::clear() {
-		if (gamedata.filename_list.empty() && !gamedata.filename.empty()) {
-			name_label_.set_text(
-			   as_richtext(as_heading_with_content(_("Map Name:"), gamedata.mapname, style_, true)));
-
-			// Show game information
-			std::string description = as_heading_with_content(
-			   mode_ == Mode::kReplay ?
-			      /** TRANSLATORS: The time a replay starts. Shown in the replay loading screen*/
-			      _("Start of Replay:") :
-			      /** TRANSLATORS: The current time of a savegame. Shown in the game saving and
-			         loading screens. */
-			      _("Game Time:"),
-			   gamedata.gametime, style_);
-
-			description = (boost::format("%s%s") % description %
-			               as_heading_with_content(_("Players:"), gamedata.nrplayers, style_))
-			                 .str();
-
-			description = (boost::format("%s%s") % description %
-			               as_heading_with_content(_("Widelands Version:"), gamedata.version, style_))
-			                 .str();
-
-			description = (boost::format("%s%s") % description %
-			               as_heading_with_content(_("Win Condition:"), gamedata.wincondition, style_))
-			                 .str();
-
-			std::string filename = gamedata.filename;
-			// Remove first directory from filename. This will be the save/ or replays/ folder
-			assert(filename.find('/') != std::string::npos);
-			filename.erase(0, filename.find('/') + 1);
-			assert(!filename.empty());
-			description = (boost::format("%s%s") % description %
-			               as_heading_with_content(_("Filename:"), filename, style_))
-			                 .str();
-
-			descr_.set_text(as_richtext(description));
-
-			std::string minimap_path = gamedata.minimap_path;
-			if (!minimap_path.empty()) {
-				try {
-					// Load the image
-					minimap_image_ = load_image(
-					   minimap_path,
-					   std::unique_ptr<FileSystem>(g_fs->make_sub_file_system(gamedata.filename)).get());
-					minimap_icon_.set_visible(true);
-					minimap_icon_.set_icon(minimap_image_.get());
-				} catch (const std::exception& e) {
-					log("Failed to load the minimap image : %s\n", e.what());
-				}
-			}
-		} else if (!gamedata.filename_list.empty()) {
-			std::string filename_list = richtext_escape(gamedata.filename_list);
-			boost::replace_all(filename_list, "\n", "<br> • ");
-			name_label_.set_text(
-			   as_richtext(as_heading_with_content(gamedata.mapname, "", style_, true)));
-
-			descr_.set_text(
-			   as_richtext(as_heading_with_content("", filename_list, style_, true, true)));
-			minimap_icon_.set_visible(false);
+	size_t number_of_files = 0;
+	size_t number_of_directories = 0;
+	for (const SavegameData& g : gamedata) {
+		if (g.is_directory()) {
+			number_of_directories++;
+		} else {
+			number_of_files++;
 		}
-	} else {
+	}
+	std::string name_list = richtext_escape(as_filename_list(gamedata));
+	boost::replace_all(name_list, "\n", "<br> • ");
+
+	std::string combined_header = as_richtext(as_heading_with_content(
```

Comment:
It goes something like this:

```
const std::string header_part_2( (boost::format(ngettext("%d file:", "%d files:", number_of_files)) % number_of_files)
	         .str());	
std::string combined_header = as_richtext(as_heading_with_content(
	   (boost::format(ngettext(
	       "Selected %1% directory and %2%", "Selected  %1% directories and %2%", number_of_directories)) %
	    number_of_directories % header_part_2)
	         .str(),
	   "", style_, true));
```
Add TRANSLATORS comments too so that translators will know how the string will be assembled.